ocaml/search.ml

```diff
@@ -0,0 +1,43 @@
+let fold_lines f init file =
+  let input = open_in file in
+  let rec fold state =
+    match input_line input with
+    | line -> fold (f state line)
+    | exception _ -> close_in input; state in
+  fold init
+
+open Batteries
+
+let knicks_re =
+  Re_pcre.re "^.*\t(.*)\t.*\t.*knicks.*$" |> Re.no_case |> Re.whole_string |> Re.compile
```

Comment:
Seems like you are using the capture group in this regex to grab the hood _and_ test for the presence of `knicks`. That's a slight variation from the other implementations but makes complete sense.